app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx

```diff
@@ -15,73 +15,207 @@ import SvgMask from '#/components/SvgMask'
 // ==============
 
 /** Props for a {@link Button}. */
-export interface ButtonProps extends Readonly<aria.ButtonProps> {
+export type ButtonProps =
+  | (aria.ButtonProps & BaseButtonProps & PropsWithoutHref)
+  | (aria.LinkProps & BaseButtonProps & PropsWithHref)
+
+/**
+ * Props for a button with an href.
+ */
+interface PropsWithHref {
+  readonly href: string
+}
+
+/**
+ * Props for a button without an href.
+ */
+interface PropsWithoutHref {
+  readonly href?: never
+}
+
+/**
+ * Base props for a button.
+ */
+export interface BaseButtonProps {
   readonly loading?: boolean
-  readonly variant: 'cancel' | 'delete' | 'icon' | 'submit'
+  /**
+   * The variant of the button
+   */
+  readonly variant: Variant
+  /**
+   * The icon to display in the button
+   */
   readonly icon?: string
-  /** FIXME: This is not yet implemented
+  /**
+   * The size of the button
+   */
+  readonly size?: Size
+  readonly rounding?: Roundings
+  /**
+   * Button takes the full width of its container
+   */
+  readonly fullWidth?: boolean
+  /**
    * The position of the icon in the button
-   * @default 'start' */
-  readonly iconPosition?: 'end' | 'start'
+   * @default 'start'
+   */
+  readonly iconPosition?: IconPosition
 }
 
+/**
+ *
+ */
+export type Roundings = 'full' | 'large' | 'medium' | 'none' | 'small' | 'xlarge'
+
+/**
+ * The size of the button
+ */
+export type Size = 'custom' | 'hero' | 'large' | 'medium' | 'small' | 'xsmall' | 'xxsmall'
+
+/**
+ * The position of the icon in the button
+ */
+export type IconPosition = 'end' | 'start'
+/**
+ * The variant of the button
+ */
+export type Variant =
+  | 'cancel'
+  | 'custom'
+  | 'delete'
+  | 'icon'
+  | 'link'
+  | 'outline'
+  | 'primary'
+  | 'submit'
+  | 'tertiary'
+
 const DEFAULT_CLASSES =
-  'flex cursor-pointer rounded-sm border border-transparent transition-[opacity,outline-offset] duration-200 ease-in-out'
+  'flex whitespace-nowrap cursor-pointer border border-transparent transition-[opacity,outline-offset] duration-150 ease-in-out select-none text-center items-center justify-center'
```

Comment:
should probably just be `duration` to use the default, although it seems like the default is 100ms

Reply:
Here's the answer. I want 150 ms